packages/eas-cli/src/commands/device/delete.ts

```diff
@@ -0,0 +1,184 @@
+              type: 'select',
+              name: 'appleTeamIdentifier',
+              message: 'What Apple Team would you like to list devices for?',
+              choices: teams.map(team => ({
+                title: team.appleTeamName
+                  ? `${team.appleTeamName} (ID: ${team.appleTeamIdentifier})`
+                  : team.appleTeamIdentifier,
+                value: team.appleTeamIdentifier,
+              })),
+            });
+
+            appleTeamIdentifier = result.appleTeamIdentifier;
+          }
+        } else {
+          teamSpinner.fail(`Couldn't find any teams for the account ${accountName}`);
+        }
+      } catch (e) {
+        teamSpinner.fail(`Something went wrong and we couldn't fetch the list of teams`);
+        throw e;
+      }
+    }
+
+    assert(appleTeamIdentifier, 'No team identifier is specified');
+
+    const devicesSpinner = ora().start('Fetching the list of devices for the team…');
+
+    try {
+      const result = await AppleDeviceQuery.getAllForAppleTeamAsync(
+        accountName,
+        appleTeamIdentifier
+      );
+
+      if (result?.appleDevices.length) {
+        const { appleTeamName, appleDevices } = result;
+
+        let chosenDevices: (AppleDeviceQueryResult | AppleDevice)[] = [];
+
+        devicesSpinner.succeed(
+          `Found ${appleDevices.length} devices for team ${appleTeamName ?? appleTeamIdentifier}`
+        );
+
+        Log.newLine();
+        if (udids) {
+          udids.forEach(udid => {
+            const foundDevice = appleDevices.find(device => device.identifier === udid);
+            if (foundDevice) {
+              chosenDevices.push(foundDevice);
+            } else {
+              Log.warn(`No device found with UDID ${udid}.`);
+            }
+          });
+        }
+
+        if (chosenDevices.length === 0) {
+          Log.addNewLineIfNone();
+          chosenDevices = await chooseDevicesToDeleteAsync(appleDevices);
+          Log.newLine();
+        }
+
+        Log.addNewLineIfNone();
+        chosenDevices.forEach(device => {
+          Log.log(
+            `${formatDevice(device, {
+              appleTeamName,
+              appleTeamIdentifier: appleTeamIdentifier!,
+            })}`
+          );
+          Log.newLine();
+        });
+
+        if (chosenDevices.length > 0) {
+          Log.warn(
+            `You are about to remove the Apple device${
+              chosenDevices.length > 1 ? 's' : ''
+            } listed above from your Expo account.`
+          );
+          Log.newLine();
+
+          const confirmed = await toggleConfirmAsync({
+            message: 'Are you sure you wish to proceed?',
+          });
+
+          if (confirmed) {
+            const removalSpinner = ora(`Removing Apple devices on Expo`).start();
+            try {
+              chosenDevices.forEach(async chosenDevice => {
+                await AppleDeviceMutation.deleteAppleDeviceAsync(chosenDevice.id);
+              });
```

Comment:
```suggestion
              await Promise.all(chosenDevices.map(chosenDevice => {
                 return AppleDeviceMutation.deleteAppleDeviceAsync(chosenDevice.id);
              }));
```
you are not awaiting for those values, forEach works only with synchronous methods (alternatively you could use for loop if it has to be called one by one)

plus nit pick: Irecomend moving `removalSpinner.succeed` in the try  block at the end

Reply:
Reading up on this, I will actually turn this into a normal for loop, as I cannot control the degree of parallelism and I don't want to run into weird errors if a user decides to remove 20 devices at the same time.